File: 백준/Silver/11047. 동전 0/동전 0.java
import java.util.*;
import java.io.*;

public class Main{

    public static void main(String[] args) throws IOException {
        int n, k, temp;
        int res = 0, start = 0;
        List<Integer> v = new ArrayList<>();

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        k = Integer.parseInt(st.nextToken());

        for(int i = 0 ; i < n ; i++){
            temp = Integer.parseInt(br.readLine());
            v.add(temp);
            if(temp <= k){
                start++;
            }
        }

        start--;
        while(k > 0){
            if(k - v.get(start) >= 0){
                res++;
                k -= v.get(start);
            }
            else{
                start--;
            }
        }

        System.out.println(res);
    }
}

File: 백준/Silver/11047. 동전 0/동전 0.cc
#include <bits/stdc++.h>
using namespace std;

int n, k, temp,res, start;
vector<int> v;

int main(){
	
	cin >> n >> k;
	for(int i = 0 ; i < n ; i++){
		cin >> temp;
		v.push_back(temp);
		if(temp <= k){
			start++;
		}
	}
	
	start--;
	while(k > 0){
		if(k - v[start] >= 0) {
			res++;
			k -= v[start];
		}
		else{
			start--;
		}
	}
	
	cout << res << "\n";
	
	return 0;
}